### PR TITLE
Add API error handling and bump to 0.1.0

### PR DESCRIPTION
**Pull request by acraileanu**

### CODE REVIEW

File: lib/passfort/errors/chargeable_limit_reached_error.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Passfort
+  module Errors
+    # Specific error class for when the chargeable limit has been reached
+    class ChargeableLLimitReachedError < APIError
```
Comment (hmac) [inline lib/passfort/errors/chargeable_limit_reached_error.rb]: you have a double L here

File: spec/passfort/http_spec.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+RSpec.describe Passfort::Http do
+  shared_examples "an error response" do
+    before do
+      stub_request(method, api_path).
+        to_return(status: status, body: body)
+    end
+
+    # rubocop:disable RSpec/MultipleExpectations
+    it "raises a specific APIError" do
+      is_expected.to raise_error do |error|
```
Comment (hmac) [inline spec/passfort/http_spec.rb]: Oh actually you can do with without multiple expectations:
```ruby
is_expected.to raise_error(instance_of(error_class))
```

File: spec/passfort/http_spec.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+      it "includes the id response" do
+        expect(result).to include("id" => "6c1d594a-496e-11e7-911e-acbc32b67d7b")
+      end
+    end
+  end
+
+  let(:http) { described_class.new("api_key") }
+  let(:path) { "a/path" }
+  let(:api_path) { Passfort::Http::DOMAIN + Passfort::Http::ROOT_PATH + path }
```
Comment (hmac) [inline spec/passfort/http_spec.rb]: These are used by  the preceding tests so I'd put them at the top.

File: spec/passfort/http_spec.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    # rubocop:enable RSpec/MultipleExpectations
+  end
+
+  shared_examples "an API call that handles all errors" do
+    context "when returning a request error" do
+      let(:status) { 404 }
+      let(:body) { "{}" }
+      let(:error_class) { Passfort::Errors::RequestError }
+
+      it_behaves_like "an error response"
```
Comment (hmac) [inline spec/passfort/http_spec.rb]: Thought I'd point out that you can write all these tests without a shared example in the same or fewer lines of code
```ruby
before { stub_request(method, api_path).to_return(status: status, body: body) }

context "when returning a request error" do
  let(:status) { 404 }
  let(:body) { "{}" }
  let(:error_class) { Passfort::Errors::RequestError }
  
  it { is_expected.to raise_error(instance_of(error_class)) }
end

context "when returning an invalid API Key error" do
  ...
end
```

File: spec/passfort/http_spec.rb (hunk 2 above)
Comment (acraileanu) [reply spec/passfort/http_spec.rb]: 👍

File: spec/passfort/http_spec.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+RSpec.describe Passfort::Http do
+  shared_examples "an error response" do
+    before do
+      stub_request(method, api_path).
+        to_return(status: status, body: body)
+    end
+
+    it "raises a specific APIError" do
+      is_expected.to raise_error(instance_of(error_class))
```
Comment (hmac) [inline spec/passfort/http_spec.rb]: I would expect us to need to define a `subject` for this to work...

File: spec/passfort/http_spec.rb [hunk 6]
```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe Passfort::Http do
+  shared_examples "an error response" do
```
Comment (hmac) [inline spec/passfort/http_spec.rb]: Not a huge fan of having one shared example that uses another shared example. It's a bit confusing...

File: lib/passfort/version.rb [hunk 7]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module Passfort
-  VERSION = "0.0.2"
+  VERSION = "0.0.3"
```
Comment (hmac) [inline lib/passfort/version.rb]: I reckon this warrants at least a minor version bump

File: spec/passfort/http_spec.rb (hunk 5 above)
Comment (acraileanu) [reply spec/passfort/http_spec.rb]: The subject is defined in the calling tests, e.g.:
```subject { -> { http.get(path) } }```

**Comment by acraileanu** [comment]
I've re-arranged the tests. Can you please have a look?

**Pull request merged**